token/client/src/token.rs

```diff
@@ -765,23 +768,39 @@ where
     }
 
     /// Close account into another
-    pub async fn close_account<S: Signer>(
+    pub async fn close_account<S: Signers>(
         &self,
         account: &Pubkey,
         destination: &Pubkey,
-        authority: &S,
+        authority: &Pubkey,
+        signing_keypairs: &S,
     ) -> TokenResult<T::Output> {
-        self.process_ixs(
-            &[instruction::close_account(
-                &self.program_id,
-                account,
-                destination,
-                &authority.pubkey(),
-                &[],
-            )?],
-            &[authority],
-        )
-        .await
+        // XXX i know this is stupid. but maybe it would feel less stupid buried in a helper function!
+        let signer_pubkeys = signing_keypairs.pubkeys();
+        let signer_pubkey_refs: &[&Pubkey] = &signer_pubkeys
+            .iter()
+            .map(Borrow::borrow)
+            .collect::<Vec<&Pubkey>>();
+
+        let mut instructions = vec![instruction::close_account(
+            &self.program_id,
+            account,
+            destination,
+            authority,
+            signer_pubkey_refs,
+        )?];
```

Comment:
this works tho i think

Reply:
This API is a bit tricky -- the idea is that the last parameter `signer_pubkeys` provides the signers *of the multisig*. So the last parameter should only be used in a multisig situation.

Ideally, the `authority` parameter would either be a Signer or a Pubkey, but that's tricky to do.

As an alternative, we could check the signers, and if there's just one whose pubkey matches the authority, then we pass empty, otherwise we pass them all in.  What do you think?